src/wp-includes/class-wp-scripts.php

```diff
@@ -512,31 +476,124 @@ public function add_inline_script( $handle, $data, $position = 'after', $standal
 	 * @return string|false Script on success, false otherwise.
 	 */
 	public function print_inline_script( $handle, $position = 'after', $display = true ) {
-		$output = $this->get_data( $handle, $position );
-
+		_deprecated_function( __FUNCTION__, '6.3.0', 'WP_Scripts::get_inline_script_data() or WP_Scripts::get_inline_script_tag()' );
+		if ( $display ) {
+			$output = $this->get_inline_script_tag( $handle, $position );
+			echo $output;
+		} else {
+			$output = $this->get_inline_script_data( $handle, $position );
+		}
 		if ( empty( $output ) ) {
 			return false;
 		}
+		return $output;
+	}
 
-		$output = trim( implode( "\n", $output ), "\n" );
+	/**
+	 * Gets data for inline scripts registered for a specific handle.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle   Name of the script to add the inline script to.
+	 *                         Must be lowercase.
```

Comment:
```suggestion
	 * @param string $handle   Name of the script to get data for.
	 *                         Must be lowercase.
```

Or something similar.